src/azure/storage/signer.rs

```diff
@@ -66,20 +72,24 @@ impl Builder {
     ///
     /// The builder should not be used anymore.
     pub fn build(&mut self) -> Result<Signer> {
+        if self.use_anonymous {
```

Comment:
The API I want to expose to users is `allow_anonymous` which is different from `use_anonymous`.

- For `allow_anonymous`, users can fetch credentials from env, and if not found, allow visiting azblob as anonymous.
- For `use_anonymous`, users have to decide visiting azblob as anonymous or not during compilation which is less useful.